lib/resource/resource.ex

```diff
@@ -2172,9 +2178,11 @@ defmodule AshGraphql.Resource do
 
   defp attribute_filter_fields(resource, schema) do
     field_names = AshGraphql.Resource.Info.field_names(resource)
+    filterable_fields = AshGraphql.Resource.Info.filterable_fields(resource)
 
     resource
     |> Ash.Resource.Info.public_attributes()
+    |> Enum.filter(&(is_nil(filterable_fields) or &1.name in filterable_fields))
     |> Enum.filter(&filterable?(&1, resource))
```

Comment:
Alright, this is very close :D

Two things:
1. pull main into your branch, some stuff has changed here
2. `Enum.filter() |> Enum.filter()` will have credo yell at you, you'll need to combine them.